lib/DragDropAction.vala

```diff
@@ -214,7 +214,7 @@ namespace Gala {
                         actor.get_transformed_position (out x, out y);
 
                         // release has happened within bounds of actor
-                        if (x < ex && x + actor.width > ex && y < ey && y + actor.height > ey) {
+                        if (clicked && x < ex && x + actor.width > ex && y < ey && y + actor.height > ey) {
```

Comment:
Do we also want to ungrab_actor in this case?

Reply:
I guess so? It's already there on line 221.

Reply:
Makes sense. Fixed.